entities/src/main/java/org/odk/collect/entities/storage/InMemEntitiesRepository.kt

```diff
@@ -51,9 +52,15 @@ class InMemEntitiesRepository : EntitiesRepository {
         property: String,
         value: String
     ): List<Entity.Saved> {
-        return getEntities(list).filter { entity ->
-            entity.properties.any { (first, second) -> first == property && second == value }
-        }.toList()
+        return if (listProperties[list]?.contains(property) == true) {
+            getEntities(list).filter { entity ->
+                entity.properties.any { (first, second) -> first == property && second == value }
+            }.toList()
+        } else if (value == "") {
```

Comment:
The same question as in https://github.com/getodk/collect/pull/6351/files#diff-ec4c6523d2114ed389e544b1831a833acda3a8066b565d4fab4b1a653c90ecbaR225